tools/lint-yaml.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Function to check if the YAML file contains the specified Go version after 'GO_VERSION:'
+check_go_version() {
+    local yamlfile="$1"
+    local required_go_version="$2"
+
+    # Use grep to find lines with 'GO_VERSION:'
+    local go_lines=$(grep -i 'GO_VERSION:' "$yamlfile" || true)  # Ignore grep exit status
+
+    # Check if any lines specify the Go version
+    if [ -n "$go_lines" ]; then
+        # Extract the Go version from lines starting with 'GO_VERSION:'
+        local extracted_go_version=$(echo "$go_lines" | sed -n 's/^[[:space:]]*GO_VERSION:[[:space:]]*\(['\''"]\?\)\?\([0-9]\+\.[0-9]\+\.[0-9]\+\)\(['\''"]\?\)\?/\2/p')
```

Comment:
Regex looks ok, but not clear to me why we need this much larger regex compared to the matching line in `lint-dockerfile.sh`:

https://github.com/lightninglabs/taproot-assets/blob/ba5a7a54e1498d88bc88fda9d0224f3dec733d12/tools/lint-dockerfile.sh#L12

Reply:
I've added a comment to the script to explain what sort of strings we're hoping to match against. For `Dockerfile` files the Go version can be found after `FROM golang:` each time, without much variation. But for YAML files we need to match on:
```
GO_VERSION: "1.21.0"
GO_VERSION: '1.21.0'
GO_VERSION: 1.21.0
GO_VERSION:1.21.0
    GO_VERSION:1.21.0
```